packages/cloud_functions/cloud_functions_platform_interface/lib/cloud_functions_platform_interface.dart

```diff
@@ -0,0 +1,82 @@
+// Copyright 2020 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+library cloud_functions_platform_interface;
+
+import 'dart:async';
+
+import 'package:firebase_core/firebase_core.dart';
+import 'package:flutter/services.dart';
+import 'package:meta/meta.dart' show required, visibleForTesting;
+
+part 'src/https_callable.dart';
+part 'src/https_callable_result.dart';
+part 'src/method_channel_cloud_functions.dart';
+
+/// The interface that implementations of `cloud_functions` must extend.
+///
+/// Platform implementations should extend this class rather than implement it
+/// as `cloud_functions` does not consider newly added methods to be breaking
+/// changes. Extending this class (using `extends`) ensures that the subclass
+/// will get the default implementation, while platform implementations that
+/// `implements` this interface will be broken by newly added
+/// [CloudFunctionsPlatform] methods.
+abstract class CloudFunctionsPlatform {
+  /// Only mock implementations should set this to `true`.
+  ///
+  /// Mockito mocks implement this class with `implements` which is forbidden
+  /// (see class docs). This property provides a backdoor for mocks to skip the
+  /// verification that the class isn't implemented with `implements`.
+  @visibleForTesting
+  bool get isMock => false;
```

Comment:
This is a deprecated way of doing federation. You can find the new approach here: https://github.com/FirebaseExtended/flutterfire/pull/1670/files#diff-0f5e1382550b037f55f569cbfe8b11adR45

See also: https://pub.dev/packages/plugin_platform_interface